### ISSUE TITLE
applyNumberFormat attribute is missing, date and time formats are broken in OpenOffice

### ISSUE THREAD

**Issue by george-polevoy** [EPPlusSoftware/EPPlus#643]
After upgrading to Epplus 6 beta, files are saved without the `applyNumberFormat` attribute in formatting elements.
It breaks date and time formatting in OpenOffice and some other software packages, reading xlsx documents.

I've tried opening the files with zip and adding this attribute manually, then archiving with zip again. After this fix it works as normal in OpenOffice.

Also I've noted that cellXfs count property shows the wrong value 1 instead of the correct 11.

Correct one (some older version of epplus)
```
<?xml version="1.0" encoding="utf-8"?>
<styleSheet xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main">
    <numFmts count="3">
        <numFmt numFmtId="164" formatCode="[h]:mm" />
        <numFmt numFmtId="165" formatCode="# ##0" />
        <numFmt numFmtId="166" formatCode="0.000" />
    </numFmts>
    <fonts count="2">
        <font>
            <sz val="11" />
            <name val="Calibri" />
        </font>
        <font>
            <b />
            <sz val="11" />
            <name val="Calibri" />
        </font>
    </fonts>
    <fills count="2">
        <fill>
            <patternFill patternType="none" />
        </fill>
        <fill>
            <patternFill patternType="gray125" />
        </fill>
    </fills>
    <borders count="5">
        <border>
            <left />
            <right />
            <top />
            <bottom />
            <diagonal />
        </border>
        <border>
            <left />
            <right />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
        <border>
            <left style="thin" />
            <right />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
        <border>
            <left style="thin" />
            <right style="thin" />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
        <border>
            <left />
            <right style="thin" />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
    </borders>
    <cellStyleXfs count="1">
        <xf numFmtId="0" fontId="0" />
    </cellStyleXfs>
    <cellXfs count="11">
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" xfId="0" applyProtection="1" />
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" borderId="1" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" borderId="2" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" borderId="4" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" applyNumberFormat="1" fontId="1" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="center" />
        </xf>
        <xf numFmtId="164" applyNumberFormat="1" fontId="0" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="165" applyNumberFormat="1" fontId="0" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="2" applyNumberFormat="1" fontId="0" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="166" applyNumberFormat="1" fontId="0" applyFont="1" borderId="3" applyBorder="1" xfId="0" applyProtection="1" applyAlignment="1">
            <alignment horizontal="right" />
        </xf>
    </cellXfs>
    <cellStyles count="1">
        <cellStyle name="Normal" xfId="0" builtinId="0" />
    </cellStyles>
    <dxfs count="0" />
</styleSheet>
```

Wrong one (epplus 6 beta)
```
<?xml version="1.0" encoding="utf-8"?>
<styleSheet xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main">
    <numFmts count="3">
        <numFmt numFmtId="164" formatCode="[h]:mm" />
        <numFmt numFmtId="165" formatCode="# ##0" />
        <numFmt numFmtId="166" formatCode="0.000" />
    </numFmts>
    <fonts count="2">
        <font>
            <sz val="11" />
            <name val="Calibri" />
        </font>
        <font>
            <b />
            <sz val="11" />
            <name val="Calibri" />
        </font>
    </fonts>
    <fills count="2">
        <fill>
            <patternFill patternType="none" />
        </fill>
        <fill>
            <patternFill patternType="gray125" />
        </fill>
    </fills>
    <borders count="5">
        <border>
            <left />
            <right />
            <top />
            <bottom />
            <diagonal />
        </border>
        <border>
            <left />
            <right />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
        <border>
            <left style="thin" />
            <right />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
        <border>
            <left style="thin" />
            <right style="thin" />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
        <border>
            <left />
            <right style="thin" />
            <top style="thin" />
            <bottom style="thin" />
            <diagonal />
        </border>
    </borders>
    <cellStyleXfs count="1">
        <xf numFmtId="0" fontId="0" />
    </cellStyleXfs>
    <cellXfs count="1">
        <xf numFmtId="0" applyNumberFormat="1" fontId="0" applyFont="1" xfId="0" applyProtection="1" />
        <xf numFmtId="0" fontId="0" borderId="1">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" fontId="0" borderId="2">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" fontId="0" borderId="3">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" fontId="0" borderId="4">
            <alignment horizontal="left" />
        </xf>
        <xf numFmtId="0" fontId="1" borderId="3">
            <alignment horizontal="center" />
        </xf>
        <xf numFmtId="164" fontId="0" borderId="3">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="165" fontId="0" borderId="3">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="2" fontId="0" borderId="3">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="0" fontId="0" borderId="3">
            <alignment horizontal="right" />
        </xf>
        <xf numFmtId="166" fontId="0" borderId="3">
            <alignment horizontal="right" />
        </xf>
    </cellXfs>
    <cellStyles count="1">
        <cellStyle name="Normal" xfId="0" builtinId="0" />
    </cellStyles>
    <dxfs count="0" />
</styleSheet>
```

**Comment by JanKallman** [comment]
I'll have a look

**Comment by JanKallman** [comment]
Looking at latest version of EPPlus, 6.0.4, I can't see this behavior. Can you verify with the latest version? If you still have a problem with this, can you add some code to reproduce it?

**Comment by george-polevoy** [comment]
Here is the code I'm using.


``` C#
        var path = ".................";
        using (var p = new ExcelPackage(path))
        {
            var ws = p.Workbook.Worksheets.Add("test");

            void Set(ExcelRange r, object value, string format)
            {
                if (format != null)
                {
                    r.Style.Numberformat.Format = format;
                }

                r.Value = value;
            }

            var ts = TimeSpan.FromHours(35.3);
            // Following timespan formats should render the same time  "35:18"

            Set(ws.Cells[1, 1], ts, "[h]:mm");
            Set(ws.Cells[2, 1], ts, "[H]:MM");
            Set(ws.Cells[3, 1], ts.TotalDays, "[H]:MM");
            Set(ws.Cells[4, 1], ts.TotalDays, "[HH]:MM");
            Set(ws.Cells[5, 1], new DateTime(2022, 05, 12, 13, 35, 15), "m/d/yyyy");
            Set(ws.Cells[6, 1], new DateTime(2022, 05, 12, 13, 35, 15), "m/d/yyyy;@");
            p.Save();
        }
```

It should generate the following (as viewed in Microsoft Excel):
<img width="210" alt="image" src="https://user-images.githubusercontent.com/2792885/168290370-c34029b1-ee53-47cd-a82b-0eb0a02360a6.png">

The same file as opened in OpenOffice Calc looks like the following:
<img width="682" alt="image" src="https://user-images.githubusercontent.com/2792885/168292054-982c0936-4b3b-469a-b8a4-26e3da5b27cb.png">


After fixing the following line in code, It opens as normal in OpenOffice:
<img width="1212" alt="image" src="https://user-images.githubusercontent.com/2792885/168292268-0d288279-a63c-4d0a-bce7-43180883fdda.png">

<img width="626" alt="image" src="https://user-images.githubusercontent.com/2792885/168290585-e4c5cad6-7009-4f03-82c7-c0478990da05.png">

Here is the file demonstrating the problem:
[file.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/8688176/file.xlsx)

**Comment by JanKallman** [comment]
Yes, I can reproduce this. I'll provide a fix shortly.

**Comment by JanKallman** [comment]
This is fixed in version 5.8.11 and 6.0.5
